akka-http2-support/src/main/scala/akka/http/scaladsl/Http2.scala

```diff
@@ -183,6 +183,8 @@ final class Http2Ext(private val config: Config)(implicit val system: ActorSyste
 }
 
 object Http2 extends ExtensionId[Http2Ext] with ExtensionIdProvider {
+  val streamId = AttributeKey[Int]("x-http2-stream-id")
```

Comment:
Maybe rename to `StreamIdAttr` or `StreamIdAttribute`?

Not sure that is best place for it but as long as it is still internal, it doesn't really matter.

Reply:
This looks so repetitive:

    response.attribute(Http2.streamIdAttribute)

... maybe:

    response.attribute(`x-http2-stream-id`)

?

Reply:
Not sure, that looks as if we'd like to pretend that the attribute is still a header...

Then maybe just keep it as `streamId`.

Reply:
Oh, I kindof liked the idea ;) - but no big deal since it's internal anyway.

> Not sure that is best place for it

Agreed, but didn't see an obvious place for it in `akka.http.impl.engine.http2` either